src/util/randomnum.js

```diff
@@ -0,0 +1,4 @@
+// Home 화면에서 보여줄 상품 리스트 랜덤으로 4개 보여주기
+export default function randomNum() {
+  return Math.floor(Math.random() * 96);
+}
```

Comment:
`randomNum(max = 4)` 이렇게 인자 값을 주면 해당 값 만큼 리스트를 보여줄 수 있지 않을까요? 96은 무슨 뜻인가요?

Reply:
현재 받아오는 총 데이터의 길이가 100개인데, 메인 페이지에서 보여주는 상품이 총 4개입니다. 그래서 reload 마다 새로운 상품들을 보여주고 싶어서 0~96번을 무작위로 선정해서 slice하려는 의도였습니다!

Reply:
저 같으면 randomNum를 `getRandomProductList(data)` 이렇게 고칠 것 같네요. 


```suggestion
export default function getRandomProductList(data) {
  return items[items.length * Math.random() | 0]
}
```

Reply:
! 다음과 같이 수정해보겠습니다!

Reply:
+ 생각해볼점은, 지금은 4개이지만, 만약 요구사항이 바뀌여서 10개를 랜덤으로 선택된다고 하면 어떻게 대응해야할지 확장성에 대해서도 고민해보시면 좋을 것 같아요

Reply:
처음의 수정 방향은 random으로 뽑은 숫자가 배열 데이터의 길이 - 4보다 클시 주어진 배열에서 [99,0,1,2]식으로 뽑아낼 방식으로 수정하려 했다. 구현한 코드는 다음과 같습니다.
```js
export default getRandomForSlice(data){
  const random = Math.floor(Math.random() * 100);
  const { length } = data;
  const sliceData = [];
  for (let i = random; i < random + 4; i++) {
    if (i > length - 1) {
      sliceData.push({...data[i - length]});
    } else {
      sliceData.push({...data[i]});
    }
  }
  return sliceData;
}
```

이렇게 해서 새로운 배열을 만들고 이를 state로 관리 할때 문제가 발생하였다. bookmark가 될시 별 마크의 색이 변화해야 하는데, 기존의 state형태를 고정으로 가지고 있기에 redux와 연동된 새로운 데이터를 받아오지 못하고 setState를 자식 컴포넌트에게 넘겨 수정해주어야 했다. 이는 불필요한 작업이라고 느껴서 기존의 방식처럼 number 데이터를 넘겨주는 형식을 사용하는 것으로 채택하였다.
만약 이렇게 뽑아온 배열을 state로 관리하지 않을시 북마크 클릭시 모든 상품 리스트가 재렌더링 되기 때문에, 사용 측면에서 불편함을 경험했습니다.

최종 수정 code는 다음과 같이 변경했습니다.
```js
export default function getRandomForSlice(data, chooseNum) {
  return Math.floor(Math.random() * data.length - chooseNum);
}

```
- 뿌려줄 데이터의 개수가 변경, 혹은 전체 데이터의 개수가 변경될 것을 고려했습니다. 
- 렌더링 할 데이터의 개수가 변화할 경우 Home에서 상수로 관리하고 있는 변수만 조절할 경우 자동으로 렌더링 되도록 수정했습니다.